src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -142,14 +143,25 @@ def wrapped(func):
 
 class CliCommand(object):
 
-    def __init__(self, name, handler, description=None, table_transformer=None):
+    def __init__(self, name, handler, description=None, table_transformer=None,
+                 arguments_loader=None, description_loader=None):
         self.name = name
         self.handler = handler
-        self.description = description
         self.help = None
+        self.description = description_loader() \
+                           if description_loader and self._should_load_description() \
+                           else description
         self.arguments = {}
+        self.arguments_loader = arguments_loader
         self.table_transformer = table_transformer
 
+    def load_arguments(self):
+        if self.arguments_loader:
+            self.arguments.update(self.arguments_loader())
+
+    def _should_load_description(self):
+        return not APPLICATION.session['completer_active']
```

Comment:
Doesn't whether the description needs to be loaded depend on whether help is requested and not whether a completer is active?